pkg/microservice/aslan/core/common/service/environment.go

```diff
@@ -256,18 +257,46 @@ func ListWorkloadsInEnv(envName, productName, filter string, perPage, page int,
 	if err != nil {
 		return count, resp, err
 	}
-	fillServiceDisplayName(resp, productInfo)
+
+	fillServiceInfo(resp, productInfo)
 	return count, resp, nil
 }
 
 type FilterFunc func(services []*Workload) []*Workload
 
 type workloadFilter struct {
-	Name string `json:"name"`
+	Name            string      `json:"name"`
+	ServiceName     string      `json:"serviceName"`
+	ServiceNameList sets.String `json:"-"`
+}
+
+func (f *workloadFilter) UnmarshalJSON(data []byte) error {
+	type wfAlias workloadFilter
```

Comment:
Put type definition at the package level?

Reply:
this is only alias for workloadFilter to avoid circular calls